### ISSUE TITLE
Login failed, no EULA present when using Adnroid app

### ISSUE THREAD

**Issue by ra-kal** [lendy007/homeassistant-skodaconnect#142]
Hi,

I have troubles with the login. I'm using android app and there is no EULA screen. I tried to delete app data and login and again no EULA screen. The connection is working fine in the app and using web browser. The credentials are correct.

Note that I've hidden my email and replaced it with xxx@xxx.cz in the logs

```
2022-10-17 17:20:25.469 DEBUG (MainThread) [custom_components.skodaconnect.config_flow] Creating connection to Skoda Connect
2022-10-17 17:20:25.469 INFO (MainThread) [skodaconnect.connection] Init Skoda Connect library, version 1.1.26
2022-10-17 17:20:25.469 DEBUG (MainThread) [skodaconnect.connection] Using service https://msg.volkswagen.de
2022-10-17 17:20:25.471 INFO (MainThread) [skodaconnect.connection] Initiating new login
2022-10-17 17:20:25.471 DEBUG (MainThread) [skodaconnect.connection] Starting authorization process for client skoda
2022-10-17 17:20:26.196 DEBUG (MainThread) [skodaconnect.connection] Got redirect to signin-service
2022-10-17 17:20:26.670 DEBUG (MainThread) [skodaconnect.connection] Start authorization for user xxx@xxx.cz
2022-10-17 17:20:27.083 DEBUG (MainThread) [skodaconnect.connection] Found dynamic credentials form, extracting attributes
2022-10-17 17:20:27.084 DEBUG (MainThread) [skodaconnect.connection] JSON: {'clientLegalEntityModel': {'clientId': 'f9a2359a-b776-46d9-bd0c-db1904343117@apps_vw-dilab_com', 'clientAppName': 'ŠKODA App & Portal', 'clientAppDisplayName': 'ŠKODA App & Portal', 'legalEntityInfo': {'name': 'ŠKODA', 'shortName': 'SKODA', 'productName': 'ŠKODA ID', 'theme': 'skoda', 'defaultLanguage': 'cs', 'termAndConditionsType': 'DEFAULT', 'legalProperties': {'revokeDataContact': 'DPOffice@skoda-auto.cz', 'imprintText': 'IMPRINT', 'countryOfJurisdiction': 'CZ'}}, 'imprintTextKey': 'imprint.link.text'}, 'template': 'loginAuthenticate', 'hmac': '592b77ee9d21b1e377553fc4708cc20169544979103788a6734937c5ca96e7fe', 'useClientRendering': True, 'titleKey': 'title.login.password', 'emailPasswordForm': {'@class': 'com.volkswagen.identitykit.signin.web.flows.base.model.EmailPasswordForm', 'email': 'xxx@xxx.cz', 'password': None}, 'error': None, 'relayState': '5b6d96b77fac5e7d57dc3e00be1bb4e7b4bdded3', 'nextButtonDisabled': False, 'enableNextButtonAfterSeconds': 0, 'postAction': 'login/authenticate', 'identifierUrl': 'login/identifier'}
2022-10-17 17:20:27.084 DEBUG (MainThread) [skodaconnect.connection] Finalizing login
2022-10-17 17:20:27.608 WARNING (MainThread) [skodaconnect.connection] Login failed, the terms and conditions might have been updated and need to be accepted. Login to https://www.skoda-connect.com and accept the new terms before trying again
2022-10-17 17:20:27.609 ERROR (MainThread) [custom_components.skodaconnect.config_flow] Login failed with error: The terms and conditions must be accepted first at "https://www.skoda-connect.com/"
```

**Comment by Farfar** [comment]
Can you try the workaround mentioned in the readme and login to the app on another android device? You can try www.android-x86.org for example.

**Comment by ra-kal** [comment]
So I installed the Skoda app on another mobile and successfully logged in. There was no EULA notice either. I'm still not able to add the integration, because login fails.

**Comment by Dejvic** [comment]
Hello,
i have the same issue, i tried two phone and web login and nothing terms condition not required accept.

Thanks for help.

Logger: custom_components.skodaconnect.config_flow
Source: custom_components/skodaconnect/config_flow.py:99
Integration: Skoda Connect (documentation, issues)
First occurred: 23:03:45 (1 occurrences)
Last logged: 23:03:45

Login failed with error: The terms and conditions must be accepted first at "https://www.skoda-connect.com/"

**Comment by rds76** [comment]
same issue here

**Comment by koleo9am** [comment]
I have registered to skoda-connect as a new user. Installed the mySkoda app and went successfully connected via android, car and browser on PC.
Still get the same error.

**Comment by MirekHruska** [comment]
Same problem for me:
2022-11-10 12:07:46.404 WARNING (MainThread) [skodaconnect.connection] Login failed, the terms and conditions might have been updated and need to be accepted. Login to https://www.skoda-connect.com and accept the new terms before trying again
2022-11-10 12:07:46.414 ERROR (MainThread) [custom_components.skodaconnect.config_flow] Login failed with error: The terms and conditions must be accepted first at "https://www.skoda-connect.com/"

App reinstall / new install on another android device / new account ... nothing solved that.

**Comment by koleo9am** [comment]
I sucessfully used the workaround using BlueStack 5 - Androd Nougat 64-bit. In the MySkoda app in older android the EULA appeared to be confirmed.
There is possible some other clientId GUID for MySkoda app on newer android systems.
I have Android 13 on the phone.

**Comment by Farfar** [comment]
> 

This is good input. When I get some free time I'll debug the app and update the client id's used in the integration. There's probably new IDs used in the later versions of the app hence the EULA won't show for the old client.

**Comment by Farfar** [comment]
I've tried to bypass SSL pinning on MySkida app 5.x.x but no luck so far.
The older version, which library is based on, is 4.2.0 and the latest I've been able to intercept messages from.
To be able to implement this change I would need login traces from version 5 of the app so I can update client it's.

**Comment by Zavesky** [comment]
Is there any old android emulator for pc? I have this problem too (Samsung Note10).

**Comment by koleo9am** [comment]
> Is there any old android emulator for pc? I have this problem too (Samsung Note10).

Google "Bluestack 5".

**Comment by Zavesky** [comment]
Great! It's working.... Thanks

**Comment by Farfar** [comment]
I have updated the underlying library to use same client IDs as the latest Android app, hopefully it will be easier now to accept EULA for this integration.
In the future I'm hoping to add a feature for accepting new terms or give consent when needed. Right now not possible though since I don't have enough data to go on.
I've added more debugging though so it's easier to determine root cause of issue when consent is missing or eula must be accepted.
